awx/ui_next/src/components/Lookup/MultiCredentialsLookup.jsx

```diff
@@ -0,0 +1,150 @@
+              id: cred.id,
+              key: cred.id,
+              kind: cred.kind,
+              type: cred.namespace,
+              value: cred.name,
+              label: cred.name,
+              isDisabled: false,
+            };
+            credentialTypes.push(cred);
+          }
+        });
+      });
+      this.setState({ credentialTypes });
+    } catch (err) {
+      onError(err);
+    }
+  }
+
+  async loadCredentials(params) {
+    const { selectedCredentialType } = this.state;
+    params.credential_type = selectedCredentialType.id || 1;
+    return CredentialsAPI.read(params);
+  }
+
+  toggleCredentialSelection(newCredential) {
+    const { onChange, credentials: credentialsToUpdate } = this.props;
+
+    let newCredentialsList;
+    const isSelectedCredentialInState =
+      credentialsToUpdate.filter(cred => cred.id === newCredential.id).length >
+      0;
+
+    if (isSelectedCredentialInState) {
+      newCredentialsList = credentialsToUpdate.filter(
+        cred => cred.id !== newCredential.id
+      );
+    } else {
+      newCredentialsList = credentialsToUpdate.filter(
+        credential =>
+          credential.kind === 'vault' || credential.kind !== newCredential.kind
+      );
+      newCredentialsList = [...newCredentialsList, newCredential];
+    }
+    onChange(newCredentialsList);
+  }
+
+  handleCredentialTypeSelect(value, type) {
+    const { credentialTypes } = this.state;
+    const selectedType = credentialTypes.filter(item => item.label === type);
+    this.setState({ selectedCredentialType: selectedType[0] });
+  }
+
+  render() {
+    const { selectedCredentialType, credentialTypes } = this.state;
+    const { tooltip, i18n, credentials } = this.props;
+    return (
+      <FormGroup label={i18n._(t`Credentials`)} fieldId="org-credentials">
+        {tooltip && (
+          <Tooltip position="right" content={tooltip}>
+            <QuestionCircleIcon />
+          </Tooltip>
+        )}
+        {credentialTypes && (
+          <Lookup
+            selectCategoryOptions={credentialTypes}
+            selectCategory={this.handleCredentialTypeSelect}
+            selectedCategory={selectedCredentialType}
+            onToggleItem={this.toggleCredentialSelection}
+            onloadCategories={this.loadCredentialTypes}
+            id="org-credentials"
+            lookupHeader={i18n._(t`Credentials`)}
+            name="credentials"
+            value={credentials}
+            multiple
+            onLookupSave={() => {}}
+            getItems={this.loadCredentials}
+            qsNamespace="credentials"
+            columns={[
+              {
+                name: i18n._(t`Name`),
+                key: 'name',
+                isSortable: true,
+                isSearchable: true,
+              },
+            ]}
+            sortedColumnKey="name"
+          />
+        )}
+      </FormGroup>
+    );
+  }
+}
+
+MultiCredentialsLookup.propTypes = {
+  tooltip: PropTypes.string,
```

Comment:
Make sure you get the other props documented here: `onError`, `credentials`, `onChange`